packages/devtools_app/lib/src/service/editor/editor_server.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2024 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'api_classes.dart';
+
+/// A base for classes that can act as an Editor (agnostic to the communication
+/// channel).
+abstract class EditorServer {
```

Comment:
The term "server" is a bit overloaded. Can we use something like `EditorBase` instead? TBH I am not completely understanding what an Editor Client is vs this class here, conceptually.

Reply:
I've expanded the comment here - is it clearer now?

Basically, the "Server" part is for being the editor (eg. what VS Code and IntelliJ would do) and the "Client" part is for connecting to DTD and consuming the editors services.

I added "Server" here just to try and make it a bit clearer (since "Editor" is quite vague). In this repo, we will use this class for testing and for the "Mock Editor" stager app. In theory this class could live elsewhere, but I thought it would be nice to keep the client/server together since their interfaces must match and maybe in future if we break `EditorClient` out so it could be used by other tools, it may make sense to do the same for `EditorServer` (for ex. if you're writing an editor that wants to provide services to host the sidebar).

Reply:
Seems reasonable to me. It felt nice having them together, but I agree it's a little weird having code we don't expect to use inside DevTools in there.

I had to move a few other files too from test_infra/test_data into test_infra/scenes/standalone_ui with this, as I discovered the Stager app had stopped working in Chrome. Seems like desktop can import in test_infra/scenes/... from test_infra/test_data, but web cannot (possibly because it's outside what it considered the project root?).